libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1241,12 +1241,23 @@ pair<smtutil::Expression, smtutil::Expression> SMTEncoder::arithmeticOperation(
 	auto symbMax = smt::maxValue(*intType);
 
 	smtutil::Expression intValueRange = (0 - symbMin) + symbMax + 1;
+	// y % m => x * m + k = y, k < m
+	static unsigned counter = 0;
+	string suffix = to_string(_operation.id()) + "_" + to_string(counter++);
+	smt::SymbolicIntVariable k(intType, intType, "k_" + suffix, m_context);
+	smt::SymbolicIntVariable m(intType, intType, "m_" + suffix, m_context);
+	smt::SymbolicIntVariable x(intType, intType, "x_" + suffix, m_context);
+	m_context.addAssertion(valueNoMod == (k.currentValue() + intValueRange * m.currentValue()));
+	m_context.addAssertion(k.currentValue() < intValueRange);
+	m_context.addAssertion(k.currentValue() >= 0);
```

Comment:
Actually if we just go for ``k.currentValue() >= symbMin`` and ``k.currentValue() <= symbMax``, then this should just work for signed as well, right?
And then I'd actually try to just immediately ``return {k.currentValue(), valueNoMod};`` and skip the whole ``ite`` business...

Reply:
Nice, will try that